ambassador/tests/t_retrypolicy.py

```diff
@@ -0,0 +1,75 @@
+from kat.harness import Query
+
+from abstract_tests import AmbassadorTest, HTTP, MappingTest, ServiceType
+
+
+class RetryPolicyTest(MappingTest):
+    parent: AmbassadorTest
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP(name="target")
+
+    def manifests(self) -> str:
+        s = super().manifests() or ""
+
+        return s + """
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: retry
+spec:
+  selector:
+    app: retry
+  ports:
+  - port: 3000
+    name: http
+    targetPort: http
+  type: NodePort
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: retry
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: retry
+    spec:
+      containers:
+      - name: retry
+        image: live/ambassador-retry-service:1.0.0
+        imagePullPolicy: Always
+        ports:
+        - name: http
+          containerPort: 3000
+        resources:
+          limits:
+            cpu: "0.1"
+            memory: 100Mi
+"""
+
+    def config(self):
+        yield self.target, self.format("""
+---
+apiVersion: ambassador/v1
+kind:  Mapping
+name:  {self.name}-target
+prefix: /{self.name}/retry/
+rewrite: /retry/
+service: https://{self.target.path.fqdn}
+retry_policy:
+  retry_on: "5xx"
+  num_retries: 3
+  per_try_timeout: "500ms"
+""")
+
+    def queries(self):
+        yield Query(self.parent.url("%s/retry/" % self.name), expected=200)
```

Comment:
Can you explain a bit more on how does this test the retry policy behavior? How do we know that it's retrying on `5xx`?

Reply:
i'm not sure if this test fits any requirements or best practices at all, but i'd need a bit help / guidance on implementing a solid testing logic for this one (i'm lacking a bit of experience here).

when i've added the changes, all i wanted todo was to check if it retries failed requests. so i just wrote a simple service that flips between a 503 and 200 response (first one fails, second one succeeds, and so on). with the retry policy in place, a query should always succeed on this service, therefor expecting the 200. With multiple (parallel) tests the flipping responses, as they are now, would probably cause issues though (maybe running multiple queries would be enough?). but i guess that there is a better solution to this?